File: api/zig/api_to_json.zig
const std = @import("std");

pub fn main() !void {
    const url = "https://randomuser.me/api/";
    const file_path = "data.json";

    var allocator = std.heap.page_allocator;
    const http_client = try std.net.http.Client.init(allocator);

    const response = try http_client.get(url, allocator);

    if (response.status_code == 200) {
        const body = try response.getBody(allocator);

        const file = try std.fs.cwd().createFile(file_path) catch |err| {
            std.debug.print("Error creating file: {}\n", .{err});
            return error.Err;
        };

        try file.writeAll(body);
        std.debug.print("File saved successfully.\n");
    } else {
        std.debug.print("Error requesting URL: {}\n", .{response.status_code});
    }
}

File: api/javascript/api_to_json.js
const https = require('https');
const fs = require('fs');

const url = 'https://randomuser.me/api/';
const filePath = 'data.json';

https.get(url, (response) => {
  let body = '';

  response.on('data', (chunk) => {
    body += chunk;
  });

  response.on('end', () => {
    fs.writeFile(filePath, body, (err) => {
      if (err) {
        console.error('Error writing file:', err);
      } else {
        console.log('File saved successfully.');
      }
    });
  });
}).on('error', (err) => {
  console.error('Error requesting URL:', err.message);
});
